Projects/Algorithmic-Trading/TS/Bot-Modules/Trading-Bot/Low-Frequency-Trading/TradingSimulation.js

```diff
@@ -186,6 +186,41 @@ exports.newAlgorithmicTradingBotModulesTradingSimulation = function (processInde
                 tradingSystemModuleObject.mantain()
                 tradingEpisodeModuleObject.mantain()
                 tradingEngineModuleObject.mantain()
+                /*
+                fetchBalance() from exchange prior to cycles every run
+                - Preferable to store raw data from promise since data returned may vary
+                - Not best case to store it at an UDV. Would rather see it stored at tradingCurrent->exchangeBalances->initial and current
+                - Not sure if sessionParameters is the way to proc it. An optional node could be added at the tradingSession 'Fetch Actual Balances' -> 'Interval'
+                */
+               if (sessionParameters.userDefinedParameters.config.fetchBalance !== undefined) {
+                    if (sessionParameters.userDefinedParameters.config.fetchBalance === true) {
+                        let minuteInterval = 1
+                        if (sessionParameters.userDefinedParameters.config.fetchBalanceInterval !== undefined) {
+                            minuteInterval = sessionParameters.userDefinedParameters.config.fetchBalanceInterval
+                        }
+                        let lastFetch = tradingEngine.tradingCurrent.tradingEpisode.userDefinedVariables.userDefinedVariable[0].value === 0 ? 0 : Number.parseInt(tradingEngine.tradingCurrent.tradingEpisode.userDefinedVariables.userDefinedVariable[0][1].info.time)
+                        if (Date.now() > lastFetch+Math.ceil(60000 * minuteInterval)) {
+                            let balances
+                            let exchangeAPIModuleObject = TS.projects.algorithmicTrading.botModules.exchangeAPI.newAlgorithmicTradingBotModulesExchangeAPI(processIndex)
+                            exchangeAPIModuleObject.initialize()
+                            balances = await exchangeAPIModuleObject.fetchAllBalances()
+
+                            if (balances !== undefined || balances !== false) {
```

Comment:
Replace || with && as you want both conditions to be true to enter the function.